hooks/useRbfTransactionData/helpers.ts

```diff
@@ -0,0 +1,121 @@
+import { RbfRecommendedFees, getRawTransaction, rbf } from '../../transactions';
+import {
+  AppInfo,
+  RecommendedFeeResponse,
+  SettingsNetwork,
+  BtcTransactionData,
+  StacksNetwork,
+  StacksTransaction,
+  StxTransactionData,
+} from '../../types';
+import { microstacksToStx } from '../../currency';
+
+export type RbfData = {
+  rbfTransaction?: InstanceType<typeof rbf.RbfTransaction>;
+  rbfTxSummary?: {
+    currentFee: number;
+    currentFeeRate: number;
+    minimumRbfFee: number;
+    minimumRbfFeeRate: number;
+  };
+  rbfRecommendedFees?: RbfRecommendedFees;
+  mempoolFees?: RecommendedFeeResponse;
+  isLoading?: boolean;
+  errorCode?: 'SOMETHING_WENT_WRONG';
+};
+
+export const isBtcTransaction = (
+  transaction: BtcTransactionData | StxTransactionData,
+): transaction is BtcTransactionData => transaction?.txType === 'bitcoin';
+
+export const constructRecommendedFees = (
+  lowerName: keyof RbfRecommendedFees,
+  lowerFeeRate: number,
+  higherName: keyof RbfRecommendedFees,
+  higherFeeRate: number,
+  stxAvailableBalance: string,
+): RbfRecommendedFees => {
+  const bigNumLowerFee = BigNumber(lowerFeeRate);
+  const bigNumHigherFee = BigNumber(higherFeeRate);
+
+  return {
+    [lowerName]: {
+      enoughFunds: bigNumLowerFee.lte(BigNumber(stxAvailableBalance)),
+      feeRate: microstacksToStx(bigNumLowerFee).toNumber(),
+      fee: microstacksToStx(bigNumLowerFee).toNumber(),
+    },
+    [higherName]: {
+      enoughFunds: bigNumHigherFee.lte(BigNumber(stxAvailableBalance)),
+      feeRate: microstacksToStx(bigNumHigherFee).toNumber(),
+      fee: microstacksToStx(bigNumHigherFee).toNumber(),
+    },
+  };
+};
+
+export const sortFees = (fees: RbfRecommendedFees) =>
+  Object.fromEntries(
+    Object.entries(fees).sort((a, b) => {
+      const priorityOrder = ['highest', 'higher', 'high', 'medium'];
+      return priorityOrder.indexOf(a[0]) - priorityOrder.indexOf(b[0]);
+    }),
+  );
+
+export const calculateStxData = async (
+  transaction: StxTransactionData,
+  btcNetwork: SettingsNetwork,
+  stacksNetwork: StacksNetwork,
+  appInfo: AppInfo | null,
+  stxAvailableBalance: string,
+): Promise<RbfData> => {
+  const { fee } = transaction;
+  const txRaw: string = await getRawTransaction(transaction.txid, btcNetwork);
+  const unsignedTx: StacksTransaction = deserializeTransaction(txRaw);
+
+  const [slow, medium, high] = await estimateTransaction(unsignedTx.payload, undefined, stacksNetwork);
+
+  let feePresets: RbfRecommendedFees = {};
+  let mediumFee = medium.fee;
+  let highFee = high.fee;
+  const higherFee = fee.multipliedBy(1.25).toNumber();
+  const highestFee = fee.multipliedBy(1.5).toNumber();
+
+  if (appInfo?.thresholdHighStacksFee) {
+    if (high.fee > appInfo.thresholdHighStacksFee) {
+      // adding a fee cap
+      highFee = appInfo.thresholdHighStacksFee * 1.5;
+      mediumFee = appInfo.thresholdHighStacksFee;
+    }
+  }
+
+  let minimumFee = fee.multipliedBy(1.25).toNumber();
```

Comment:
question: is it okay for the default minimumFee to have the same multiplier as the higherFee here?

Reply:
I think so, according to https://secretkeylabs.slack.com/archives/C063ZVDLCLB/p1703778799805989?thread_ts=1703712686.356469&cid=C063ZVDLCLB and https://secretkeylabs.slack.com/archives/C05QCJM9D1N/p1704972387104769?thread_ts=1704964069.991789&cid=C05QCJM9D1N